src/module.c

```diff
@@ -672,12 +671,12 @@ int moduleGetCommandKeysViaAPI(struct redisCommand *cmd, robj **argv, int argc,
     ctx.module = cp->module;
     ctx.client = NULL;
     ctx.flags |= REDISMODULE_CTX_KEYS_POS_REQUEST;
+    if (!result->keys) result->keys = result->keysbuf;  /* Initialize to use preallocated array */
```

Comment:
isn't the init with the preallocated array already handled by the caller?

Reply:
No, it can't without a macro that takes the name of the variable.

Reply:
ohh right, the caller does `GETKEYS_RESULT_INIT` which doesn't do that.
it's `prepareGetKeysResult` that does it (static and only used in db.c).
and it is designed to be used once (not repeatedly).

i guess what i was thinking is to make it non-static, and change it in a way that it can be called repeatedly, then it can be used in both `moduleGetCommandKeysViaAPI` (first init), and also in `RM_KeyAtPos` (doing the reallocs).

just a thought, your way is good too.